File: src/main/java/org/mitallast/queue/raft/protocol/Noop.java
package org.mitallast.queue.raft.protocol;

import org.mitallast.queue.common.codec.Codec;
import org.mitallast.queue.common.codec.Message;

public class Noop implements Message {
    public static final Noop INSTANCE = new Noop();
    public static final Codec<Noop> codec = Codec.of(INSTANCE);

    private Noop() {
    }
}

File: src/main/java/org/mitallast/queue/raft/protocol/Noop.kt
package org.mitallast.queue.raft.protocol

import org.mitallast.queue.common.codec.Codec
import org.mitallast.queue.common.codec.Message

class Noop private constructor() : Message {
    companion object {
        val INSTANCE = Noop()
        val codec = Codec.of(INSTANCE)
    }
}
